src/main/java/org/folio/innreach/config/SpringAsyncConfig.java

```diff
@@ -1,16 +1,46 @@
 package org.folio.innreach.config;
 
+import org.folio.innreach.domain.exception.async.SpringAsyncExceptionHandler;
 import org.springframework.aop.interceptor.AsyncUncaughtExceptionHandler;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.core.task.SimpleAsyncTaskExecutor;
 import org.springframework.scheduling.annotation.AsyncConfigurer;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
 
-import org.folio.innreach.domain.exception.async.SpringAsyncExceptionHandler;
+import java.util.concurrent.Executor;
 
 @Configuration
 public class SpringAsyncConfig implements AsyncConfigurer {
 
+  @Value("${spring.async.config.executor.pool-size}")
+  private int poolSize;
+
+  /*
+   * We need a SimpleAsyncTaskExecutor here because it will be used to run all @Async methods.
+   * Since it is a SimpleAsyncTaskExecutor, a new thread will be created for every task, and these threads will not be reused.
+   * So this is the main trick. Once created, a new thread will inherit all inheritable thread-local variables,
```

Comment:
Can you clarify this comment "So this is the main trick." I'm not sure what that is referring to.